### PR TITLE
Fix Odometry Node to Include Fake Wheel Encoder Data

### PR DESCRIPTION
**Pull request by IcedTea2K**
- [NAV136](https://trello.com/c/gA2KIMEU/136-nav136-correct-existing-odometry-node)
- A few typos were also fixed in the fake wheel encoder data
- A "test" launch file was create to test the node's behaviour

### CODE REVIEW

File: navigation/scripts/odom_publisher.py [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
+def process_encoder(data: wheel_encoder_data, is_right_wheel: bool):
+    # https://answers.ros.org/question/306140/actual-wheels-rpm-velocities-from-ticks-target-velocity-and-robot-velocity/
+    global left_rpm, right_rpm, v_lin, v_th
+    rpm = data.rpm if data.spin_direction else -data.rpm
+    if is_right_wheel:
+        right_rpm = rpm
+    else:
+        left_rpm = rpm
+    
+    v_lin = (left_rpm + right_rpm) / 120 * (fake_wheel_raduis_in_meters * 2) * np.pi
```
Comment (yel-wishahy) [inline navigation/scripts/odom_publisher.py:34]: 1 [RPM] = 2pi/60 [RAD/S] , where did the 120 come from ?

File: navigation/scripts/odom_publisher.py [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
+    v_th = ((right_rpm - left_rpm) / 120 * (fake_wheel_distance_in_meters * 2) * np.pi) / fake_wheel_distance_in_meters
```
Comment (yel-wishahy) [inline navigation/scripts/odom_publisher.py:35]: shouldn't it be fake_wheel_radius * 2?

**Review by yel-wishahy — CHANGES_REQUESTED** [review]

File: navigation/scripts/odom_publisher.py (hunk 1 above)
Comment (IcedTea2K) [reply navigation/scripts/odom_publisher.py:34]: I'm taking the average of rps of the two wheels. That's why I also divide 2 to get rps / 120

File: navigation/scripts/odom_publisher.py (hunk 2 above)
Comment (IcedTea2K) [reply navigation/scripts/odom_publisher.py:35]: Yeah, that's the one I wanted to ask you. From the source I included in the code, it's saying that if I want to calculate the angular velocity I have to use the distance between the two wheels instead. Not too sure if it's correct tho.

Comment (yel-wishahy) [reply navigation/scripts/odom_publisher.py:35]: in that case it should be 
 v_th = ((right_rpm - left_rpm) / 120 * (**fake_wheel_radius_in_meters** * 2) * np.pi) / fake_wheel_distance_in_meters

Comment (yel-wishahy) [reply navigation/scripts/odom_publisher.py:35]: right now you are dividing fake_wheel_distance_in_meters/fake_wheel_distance_in_meters which seems incorrect

File: navigation/scripts/odom_publisher.py (hunk 1 above)
Comment (yel-wishahy) [reply navigation/scripts/odom_publisher.py:34]: hmmm ,ok it looks right for linear

File: navigation/scripts/odom_publisher.py (hunk 2 above)
Comment (IcedTea2K) [reply navigation/scripts/odom_publisher.py:35]: Ah right, my bad. I didn't read the whole line of code. I'll fix it real quick

**Review by yel-wishahy — APPROVED** [review]

**Pull request merged**